packages/blocks/src/blocks/CoreImage.tsx

```diff
@@ -0,0 +1,104 @@
+import { gql } from '@apollo/client';
+import React from 'react';
+import Image from 'next/image.js';
+import { useBlocksTheme } from '../components/WordPressBlocksProvider.js';
+import { ContentBlock } from '../components/WordPressBlocksViewer.js';
+import { getStyles } from '../utils/index.js';
+
+export type CoreImageFragmentProps = ContentBlock & {
+  attributes: {
+    align?: string;
+    alt?: string;
+    anchor?: string;
+    borderColor?: string;
+    caption?: string;
+    className?: string;
+    width?: string;
+    url?: string;
+    title?: string;
+    style?: string;
+    src?: string;
+    sizeSlug?: string;
+    rel?: string;
+    lock?: string;
+    linkTarget?: string;
+    linkDestination?: string;
+    linkClass?: string;
+    id?: string;
+    href?: string;
+    height?: string;
+    cssClassName?: string;
+  };
+};
+
+export function CoreImage(props: CoreImageFragmentProps) {
+  const theme = useBlocksTheme();
+  const style = getStyles(theme, { ...props });
+  const { attributes } = props;
+
+  if (attributes.width && attributes.height) {
+    return (
+      <figure>
+        <Image
+          style={style}
+          src={attributes.src || ''}
+          width={attributes.width}
+          height={attributes.height}
+          alt={attributes.alt}
+          className={attributes?.cssClassName}
+        />
+        {attributes?.caption && <figcaption>{attributes.caption}</figcaption>}
+      </figure>
+    );
+  }
+  return (
+    <figure>
+      <img
+        style={style}
+        src={attributes.src}
+        width={attributes.width}
+        height={attributes.height}
+        alt={attributes.alt}
+        className={attributes?.cssClassName}
+      />
+      {attributes?.caption && <figcaption>{attributes.caption}</figcaption>}
+    </figure>
+  );
+}
+
+CoreImage.fragments = {
+  key: `CoreImageBlockFragment`,
+  entry: gql`
+    fragment CoreImageBlockFragment on CoreImage {
+      attributes {
+        align
+        alt
+        anchor
+        borderColor
+        caption
+        className
+        width
+        url
+        title
+        style
+        src
+        sizeSlug
+        rel
+        lock
+        linkTarget
+        linkDestination
+        linkClass
+        id
```

Comment:
Don't use an id here because Apollo with cache and reuse nodes that has the same `id` fields. This would create issues when using duplicate images in a post that have different attributes.

Instead use something like:
```
imageId: id
```

Reply:
I've added this to line 90, do I need to change line 27 as well from `id?: string;` to `ImageId?: string;` ?